### ISSUE TITLE
Micrometer-Extension writes wrong URI-Tag when Path-Variables defined at Interface-Level

### ISSUE THREAD

**Issue by lostiniceland** [quarkusio/quarkus#30843]
### Describe the bug

We are currently seing wrong metric-uri-tags for incoming requests when using the Micrometer-Prometheus extension.

Instead of placeholders for the path-variables, a metric for each path-variable-instance is created.
The issue is caused when placing the JAX-RS annotations on an interface, rather than the actual resource-class.

As a workaround, we have to duplicate `@Path` (and probably also `@PathParam`) to the implementing class, which is quite error-prone.

Unfortunatly using `quarkus.micrometer.binder.http-server.match-patterns` only work as workaround for the first appearance like `/<id>` but it fails for sub-queries like `/<id>/do`

### Expected behavior

Metric is tagged with proper uri, replacing path-variables with a placeholder
`http_server_requests_seconds_count{... uri="/{id}"} 1.0`

Path annotation should be recognized regardless of location on interface or class

### Actual behavior

A unique metric is written for each path-parameter accessing the endpoint
```
http_server_requests_seconds_count{... uri="/eae3806d-141c-46ed-a067-6f4b81f36062"} 2.0
http_server_requests_seconds_count{... uri="/eae3806d-141c-yyyy-xxxx-6f4b81f36062"} 1.0
```

### How to Reproduce?

Enable dependency `quarkus-micrometer-registry-prometheus`

Create a endpoint consisting of interface and impl.
```
Path("/")
public interface SomeApi {
   @GET
   @Path("{id}")
   Response load(@PathParam("id) String id);
}

@RequestScoped
public class SomeResource implements SomeApi {
   public Response load(String id){
   ...do
   }
}
```

Access App with `/someid` to trigger metrics and check `/q/metrics` for `http_server_requests_count`

### Output of `java -version`

17

### Quarkus version or git rev

2.15.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.3

### Additional information

Our main motivation for splitting interface and implementation is to avoid the annotation-hell due to swagger and jax-rs. The implementation in our case then has the meaningfull stuff like access-control, custom interceptors.
Furthermore the split is needed when doing contract first with OpenApi-Generators.

**Comment by ebullient** [comment]
A reproducer will be helpful here. Can you show your configured match pattern?

**Comment by lostiniceland** [comment]
I will try to put something together over the weekend.

After a little debugging it looks like [RestPathAnnotationProcessor](https://github.com/quarkusio/quarkus/blob/0506e8141e2d3fde33be906f5f9a8b3e6bb64209/extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/RestPathAnnotationProcessor.java) is responsible for the issue, because it is not setting the runtime Annotation `@QuarkusRestTemplatePath` when `@Path` is only set on the interface.

**Comment by lostiniceland** [comment]
[quarkus-30843.zip](https://github.com/quarkusio/quarkus/files/10580642/quarkus-30843.zip)

There's a working sample. 

- Run Maven `mvn quarkus:dev` 
- http://localhost:8080/hello
- http://localhost:8080/world
- http://localhost:8080/q/metrics
- See broken heaviour
- Uncomment `@Path` in ` SomeResource`
- Execute Requests again
- Metrics now contain the proper `/{id}` 

I am currently working my way through the RestPathAnnotationProcessor and I think I've found a solution to fallback checking the interface when annotation is not present on the class. Though this only works for class-level annotations. The fallback needs to apply to method-level annotations on the interface as well and I am currently stuck navigating from the interface-types `org.jboss.jandex.Type` to the method. 
The idea was to iterate over the interface-types and find the matching method which is then checked for `@Path`.  But I am not yet familiar enough with Jandex.

**Comment by ebullient** [comment]
Ok. Just as something else to attempt, I updated the API to have another method:

```
@Path("/")
public interface SomeApi {
    @GET
    @Path("{id}")
    Response get(@PathParam("id") String id);
    
    @GET
    @Path("{id}/boom")
    Response getBoom(@PathParam("id") String id);
}
```

and added a quick impl to SomeResource

```
    @Override
    public Response getBoom(String id) {
        return Response.ok().entity(new Greet("BOOM " + id)).build();
    }
```

With this match pattern: 

```
quarkus.micrometer.binder.http-server.match-patterns=/[^/]+/boom=/{id}/boom
```

I get the following output: 

```
http_server_requests_seconds_count{method="GET",outcome="SUCCESS",status="200",uri="/{id}/boom",} 3.0
http_server_requests_seconds_sum{method="GET",outcome="SUCCESS",status="200",uri="/{id}/boom",} 0.202644959
```

Not saying we shouldn't fix the detection parts, just trying to make sure that you can get the match pattern to work, because it should.

**Comment by lostiniceland** [comment]
That should work, but it would require a lot of patterns (for different POST-methods) over many services.
Hence I am quite interested in fixing the detection. 

Maybe you can give me a hint how I can access the current Jandex index for the extension. The Jandex docs specify how to index a class and navigate it, but in this case the index is probably already created, but I cannot access it from `RestPathAnnotationProcessor` (which I am still assuming to be the responsible class).

**Comment by lostiniceland** [comment]
I found a working draft though this is not yet fully fleshed out (missing tests). What do you think?

**Comment by lostiniceland** [comment]
@geoand how to backport this fix to 2.16 now?

**Comment by geoand** [comment]
We will handle it, you don't need to do anything
